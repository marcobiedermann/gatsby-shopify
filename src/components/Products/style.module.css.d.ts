declare const styles: {
  readonly products: string;
  readonly products__title: string;
};
export = styles;
