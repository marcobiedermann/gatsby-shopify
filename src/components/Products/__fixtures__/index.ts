import { ProductsProps } from '..';
import productPreview from '../../ProductPreview/__fixtures__';

const products: ProductsProps = {
  products: [productPreview],
};

export default products;
