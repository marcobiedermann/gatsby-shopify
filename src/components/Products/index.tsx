import React, { FC } from "react"
import Product, { ProductProps } from "../Product"

export interface ProductsProps {
  products: ProductProps[]
}

const Products: FC<ProductsProps> = props => {
  const { products } = props

  return (
    <ul>
      {products.map(product => (
        <li key={product.shopifyId}>
          <Product {...product} />
        </li>
      ))}
    </ul>
  )
}

export default Products
