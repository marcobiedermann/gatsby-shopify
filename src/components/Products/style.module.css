.products {
  display: grid;
  gap: 1em;
  grid-template-columns: repeat(auto-fill, minmax(18em, 1fr));
  list-style: none;
  margin: 0;
  padding: 0;
}

.products li {
  background-color: #fff;
  box-shadow: 0 0 3px rgba(0, 0, 0, 0.25);
}
