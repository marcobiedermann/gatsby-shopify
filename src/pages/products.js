import { graphql, Link } from "gatsby"
import React from "react"
import Layout from "../components/layout"

const ProductsPage = ({ data }) => (
  <Layout>
    <h1>Products</h1>
    <ul>
      {data.allShopifyProduct.edges.map(({ node }) => (
        <li key={node.shopifyId}>
          <h3>
            <Link to={`/product/${node.handle}`}>{node.title}</Link>
            {" - "}${node.priceRange.minVariantPrice.amount}
          </h3>
          <p>{node.description}</p>
        </li>
      ))}
    </ul>
  </Layout>
)

export default ProductsPage

export const query = graphql`
  {
    allShopifyProduct(sort: { fields: [title] }) {
      edges {
        node {
          title
          shopifyId
          description
          handle
          priceRange {
            minVariantPrice {
              amount
            }
          }
        }
      }
    }
  }
`
