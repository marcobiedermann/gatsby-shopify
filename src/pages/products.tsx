import { graphql, Link, PageProps, useStaticQuery } from "gatsby"
import React, { FC } from "react"
import Layout from "../components/Layout"

interface ShopifyProductQuery {
  title: string
  shopifyId: string
  description: string
  handle: string
  priceRange: {
    minVariantPrice: {
      amount: string
    }
  }
}

interface AllShopifyProductQuery {
  allShopifyProduct: {
    edges: {
      node: ShopifyProductQuery
    }[]
  }
}

const ProductsPage: FC<PageProps> = props => {
  const { location } = props
  const { allShopifyProduct } = useStaticQuery<AllShopifyProductQuery>(graphql`
    {
      allShopifyProduct(sort: { fields: [title] }) {
        edges {
          node {
            title
            shopifyId
            description
            handle
            priceRange {
              minVariantPrice {
                amount
              }
            }
          }
        }
      }
    }
  `)

  return (
    <Layout location={location}>
      <h1>Products</h1>
      <ul>
        {allShopifyProduct.edges.map(edge => (
          <li key={edge.node.shopifyId}>
            <h3>
              <Link to={`/products/${edge.node.handle}`}>
                {edge.node.title}
              </Link>
              {" - "}${edge.node.priceRange.minVariantPrice.amount}
            </h3>
            <p>{edge.node.description}</p>
          </li>
        ))}
      </ul>
    </Layout>
  )
}

export default ProductsPage
