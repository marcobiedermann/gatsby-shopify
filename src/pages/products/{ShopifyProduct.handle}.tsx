import { graphql, PageProps } from 'gatsby';
import React from 'react';
import Grid from '../../components/Grid';
import Product from '../../components/Product';

interface Source {
  srcSet: string;
  sizes: string;
  type: string;
}

type Layout = 'fixed' | 'fullWidth' | 'constrained';

interface Image {
  altText: string;
  gatsbyImageData: {
    images: {
      sources: Source[];
      fallback: {
        src: string;
        srcSet: string;
        sizes: string;
      };
    };
    layout: Layout;
    width: number;
    height: number;
  };
  id: string;
}

interface Option {
  name: string;
  id: string;
  values: string[];
}

export interface DataProps {
  shopifyProduct: {
    description: string;
    handle: string;
    images: Image[];
    options: Option[];
    priceRangeV2: {
      minVariantPrice: {
        amount: string;
        currencyCode: string;
      };
      maxVariantPrice: {
        amount: string;
        currencyCode: string;
      };
    };
    shopifyId: string;
    tags: string[];
    title: string;
    vendor: string;
  };
}

interface ProductPageProps extends PageProps<DataProps> {}

function ProductPage(props: ProductPageProps) {
  const {
    data: { shopifyProduct },
  } = props;

  return (
    <>
      <Grid>
        <Product {...shopifyProduct} />
      </Grid>
    </>
  );
}

export const query = graphql`
  query ($id: String!) {
    shopifyProduct(id: { eq: $id }) {
      description
      handle
      images {
        altText
        gatsbyImageData
        id
      }
      options {
        id
        name
        values
      }
      priceRangeV2 {
        minVariantPrice {
          amount
          currencyCode
        }
        maxVariantPrice {
          amount
          currencyCode
        }
      }
      shopifyId
      tags
      title
      vendor
    }
  }
`;

export default ProductPage;
