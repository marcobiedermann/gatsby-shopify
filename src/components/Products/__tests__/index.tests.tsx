import { render } from '@testing-library/react';
import React from 'react';
import Products from '..';
import products from '../__fixtures__';

describe('products component', () => {
  it('should render correctly', () => {
    const { container } = render(<Products {...products} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
