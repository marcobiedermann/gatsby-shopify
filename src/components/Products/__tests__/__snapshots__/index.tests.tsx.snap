// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`products component should render correctly 1`] = `
<ul>
  <li>
    <div>
      <h1>
        Products
      </h1>
    </div>
  </li>
  <li>
    <div>
      <figure>
        <a
          href="/products/product"
        >
          <img
            alt="Product Title"
            src="/path/to/image.jpg"
          />
        </a>
      </figure>
      <h3>
        <a
          href="/products/product"
        >
          Product Title
        </a>
      </h3>
      <p>
        100
         
        EUR
      </p>
    </div>
  </li>
</ul>
`;
